src/gpu_macros.h

```diff
@@ -88,10 +55,13 @@
 #define gpuEventCreate(...) GPU_CHECK_ERROR(cudaEventCreate(__VA_ARGS__))
 #define gpuGetDeviceCount(...) GPU_CHECK_ERROR(cudaGetDeviceCount(__VA_ARGS__))
 #define gpuGetLastError cudaGetLastError
+#define gpuGetDevice(...) GPU_CHECK_ERROR(cudaGetDevice(__VA_ARGS__))
+#define gpuSetDevice(...) GPU_CHECK_ERROR(cudaSetDevice(__VA_ARGS__))
+#define gpuDeviceGetAttribute(...) GPU_CHECK_ERROR(cudaDeviceGetAttribute(__VA_ARGS__))
+#define gpuDeviceAttributeWarpSize cudaDevAttrWarpSize
+#define __gpu_shfl_down(...) __shfl_down_sync(0xffffffff, __VA_ARGS__)
```

Comment:
Note sure I understand these additions?

Reply:
Hi @bwmeyers, I have merged `main_gpu_bwm` into this branch. And fixed a comment that went lost in the way.

Those macros come from another project where I took the updated macros list. These are not used now, but may be in the future. There is no harm in having them there.